tests/test_ampOffset.py

```diff
@@ -0,0 +1,110 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import unittest
+import numpy as np
+
+import lsst.utils.tests
+import lsst.afw.image as afwImage
+from lsst.obs.lsst import LsstCam
+from lsst.obs.subaru import HyperSuprimeCam
+from lsst.ip.isr.ampOffset import AmpOffsetConfig, AmpOffsetTask
+
+
+class AmpOffsetTest(lsst.utils.tests.TestCase):
+    def setUp(self):
+        self.camera = None  # `lsst.afw.cameraGeom.Camera` object
+        self.backgroundFractionSample = None  # not used for now
+        self.ampEdgeWindow = None  # not used for now
+
+    def tearDown(self):
+        del self.camera
+
+    def buildExposure(self, addBackground=False):
+        exp = afwImage.ExposureF(self.camera[0].getBBox())
+        exp.setDetector(self.camera[0])
+        amps = exp.getDetector().getAmplifiers()
+        self.values = np.linspace(-2.5, 2.5, len(amps))
+        for i, amp in enumerate(amps):
+            exp.image[amp.getBBox()] = self.values[i]
+        if addBackground:
+            exp.image.array += 100
+        return exp
+
+    def testAmpOffset(self):
+        if self.camera is None:
+            raise unittest.SkipTest("Skipping due to no camera set.")
+        exp = self.buildExposure(addBackground=False)
+        amps = exp.getDetector().getAmplifiers()
+        config = AmpOffsetConfig()
+        config.doBackground = False
+        config.doDetection = False
+        task = AmpOffsetTask(config=config)
+        task.run(exp)
+        meta = exp.getMetadata().toDict()
+        self.assertEqual(np.sum(exp.image.array), 0)
+        for amp, value in zip(amps, self.values):
+            ampName = amp.getName()
+            self.assertAlmostEqual(meta[f"LSST ISR AMPOFFSET PEDESTAL {ampName}"], value, 6)
+
+    def testAmpOffsetWithBackground(self):
+        if self.camera is None:
+            raise unittest.SkipTest("Skipping due to no camera set.")
+        exp = self.buildExposure(addBackground=True)
+        amps = exp.getDetector().getAmplifiers()
+        config = AmpOffsetConfig()
+        config.doBackground = True
+        config.doDetection = True
+        task = AmpOffsetTask(config=config)
+        task.run(exp)
+        meta = exp.getMetadata().toDict()
+        nAmps = len(amps)
+        ampNames = []
+        for amp in amps:
+            ampNames.append(amp.getName())
+        for i in range(nAmps // 2):
+            self.assertAlmostEqual(
+                meta[f"LSST ISR AMPOFFSET PEDESTAL {ampNames[i]}"],
+                -meta[f"LSST ISR AMPOFFSET PEDESTAL {ampNames[nAmps-i-1]}"],
+                7,
+            )
```

Comment:
Can this check that the measured values are only minimally different than the input pedestals?  Suggestions on JIRA.

Reply:
Done!